modules/sdp/SDP.js

```diff
@@ -34,64 +57,104 @@ export default class SDP {
             }
             media[i] = mediaI;
         }
-        const session = `${media.shift()}\r\n`;
-
-        this.isP2P = isP2P;
+        this.session = `${media.shift()}\r\n`;
         this.media = media;
-        this.raw = session + media.join('');
-        this.session = session;
+    }
 
-        // This flag will make {@link transportToJingle} and {@link jingle2media} replace ICE candidates IPs with
-        // invalid value of '1.1.1.1' which will cause ICE failure. The flag is used in the automated testing.
-        this.failICE = false;
+    /**
+     * Adds or removes the sources from the SDP.
+     *
+     * @param {Object} sourceMap - The map of the sources that are being added/removed.
+     * @param {boolean} isAdd - Whether the sources are being added or removed.
+     * @returns {Array<number>} - The indices of the new m-lines that were added/modifed in the SDP.
+     */
+    updateRemoteSources(sourceMap, isAdd = true) {
+        const updatedMidIndices = [];
+
+        for (const source of sourceMap.values()) {
+            const { mediaType, msid, ssrcList, groups } = source;
+            let idx;
+
+            if (isAdd) {
+                // For P2P, check if there is an m-line with the matching mediaType that doesn't have any ssrc lines.
+                // Update the existing m-line if it exists, otherwise create a new m-line and add the sources.
+                idx = this.media.findIndex(mLine => mLine.includes(`m=${mediaType}`) && !mLine.includes('a=ssrc'));
+                if (!this.isP2P || idx === -1) {
+                    this.addMlineForNewSource(mediaType, true);
+                    idx = this.media.length - 1;
```

Comment:
Why does the index need adjusting?

Reply:
1. For JVB, we are creating a new m-line always.
2. For P2P, we try to use an existing recvonly m-line (because of a secondary local video source) for adding the new remote source, create new one otherwise.
We have to update the relevant m-line so we adjust the m-line index accordingly.